weave

```diff
@@ -207,6 +207,26 @@ dns_arg_count() {
     fi
 }
 
+dns_extra_hosts_args() {
```

Comment:
Yes, maybe it is not really related to DNS... I will rename it to `extra_hosts_args`